controller/volume_controller.go

```diff
@@ -2948,10 +2948,14 @@ func (vc *VolumeController) checkForAutoDetachment(v *longhorn.Volume, e *longho
 	if backupVolumeName, isExist := v.Labels[types.LonghornLabelBackupVolume]; isExist && backupVolumeName != "" {
 		backupVolume, err := vc.ds.GetBackupVolumeRO(backupVolumeName)
 		if err != nil {
-			return errors.Wrapf(err, "failed to get backup volume: %v", v.Name)
+			if !datastore.ErrorIsNotFound(err) {
+				return errors.Wrapf(err, "failed to get backup volume: %v", v.Name)
```

Comment:
Prevent nested if clause.

do this instead `if err != nil && !datastore.ErrorIsNotFound(err) {`